packages/ui-kit/src/components/mermaid-lazy.tsx

```diff
@@ -0,0 +1,16 @@
+import { lazy } from 'react';
+
+const mermaidVersion = '9.3';
+
+const Mermaid = lazy(() => {
+  return new Promise((resolve) => {
+    const script = document.createElement('script');
```

Comment:
@gabriele-ct  that's similar to the idea I folllowed, too but my implementation is catering for not adding the same script multiple times and the other cases the useScript hook covered.   At least I think it's the same approach (it's practically nesting two Suspense's to make sure that the loading of the external script is not happening server-side.  One Suspense loads the client side component,  the next is loading the external script client-side. 

Here's the commit:   701421daebfa7a2348805e53543dcd346fcf35cd 

@emmenko too wild?  I think it's not using undocumented React APIs, the lazy(load) API is specified in the official React type declarations and the documentation is also explaining it in general as somthing that returns a promise that resolves to a component export.

Reply:
@nkuehn I agree, in your implementation the whole script tag generation is much more robust, I just felt something was not right conceptually in your version (the component part) as you're lazy loading a component `mermaid-client-side.tsx` that in fact is rendering immediately after downloading JUST the dependencies declared in the import statements (not waiting for the mermaid library), and it's not showning the loading spinner while downloading the mermaid library from the CDN. 
I created a video to show a sort of "slow motion" of the 2 versions
Your version:
https://drive.google.com/file/d/19X7Gy5EpDCitXvjl38GitDdkaZKFzhsW/view?usp=sharing
My version:
https://drive.google.com/file/d/1y9OGQTLaknYyiF77JIgU4GVyGQ7h5YBA/view?usp=sharing

We can discuss this tomorrow, in any case my implementation is just to try it out, it needs refactoring and better handling of script tag injection

Reply:
Yes, there is something not entirely clear to me about the way the nested Suspense works in my version. 

Needs slow-mo testing on a full build though (`yarn build && yarn serve`), the dev server you recorded has a very different bundling behavior.